terraform/modules/neptune-cluster/data.tf

```diff
@@ -0,0 +1,90 @@
+data "aws_caller_identity" "current" {
+
+}
+
+data "aws_iam_policy_document" "kms" {
+
+  statement {
+    effect = "Allow"
+
+    principals {
+      type = "AWS"
+      identifiers = [
+        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+      ]
+    }
+
+    actions = [
+      "kms:CancelKeyDeletion",
+      "kms:CreateAlias",
+      "kms:CreateGrant",
+      "kms:CreateKey",
+      "kms:Decrypt",
+      "kms:DeleteAlias",
+      "kms:DescribeKey",
+      "kms:DisableKey",
+      "kms:DisableKeyRotation",
+      "kms:EnableKey",
+      "kms:EnableKeyRotation",
+      "kms:Encrypt",
+      "kms:GenerateDataKey",
+      "kms:GenerateDataKeyPair",
+      "kms:GenerateDataKeyWithoutPlaintext",
+      "kms:GenerateDataKeyPairWithoutPlaintext",
+      "kms:GetKeyPolicy",
+      "kms:GetKeyRotationStatus",
+      "kms:ListAliases",
+      "kms:ListGrants",
+      "kms:ListKeys",
+      "kms:ListResourceTags",
+      "kms:PutKeyPolicy",
+      "kms:ReEncryptFrom",
+      "kms:ReEncryptTo",
+      "kms:RetireGrant",
+      "kms:RevokeGrant",
+      "kms:Sign",
+      "kms:ScheduleKeyDeletion",
+      "kms:TagResource",
+      "kms:UntagResource",
+      "kms:UpdateAlias",
+      "kms:UpdateKeyDescription",
+      "kms:Verify"
+    ]
+    resources = [
+      aws_kms_key.this.arn,
+      aws_kms_alias.this.arn
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    principals {
+      type = "AWS"
+      identifiers = [
+        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+      ]
+    }
+
+    actions = [
+      "kms:Encrypt",
+      "kms:Decrypt",
+      "kms:GenerateDataKey",
+      "kms:ReEncryptTo",
+      "kms:GenerateDataKeyWithoutPlaintext",
+      "kms:CreateGrant",
+      "kms:ReEncryptFrom",
+      "kms:DescribeKey"
+    ]
+    resources = [
+      aws_kms_key.this.arn
+    ]
+    condition {
+      test     = "StringEquals"
+      variable = "kms:ViaService"
+      values = [
+        "rds.us-east-1.amazonaws.com"
```

Comment:
we should create a variable for the AWS region with the default of us-east-1 and reference this rather than hardcoding the region